src/cbmc/cbmc_solvers.cpp

```diff
@@ -88,22 +88,37 @@ smt2_dect::solvert cbmc_solverst::get_smt2_solver_type() const
   return s;
 }
 
+/// Create satcheck object based on options object
+/// \param options Options object
+/// \param use_simplifier If set to true, creates satchecker with simplifier
+/// \param message_handler message handler for satchecker
+/// \return Unique pointer holding a satcheck instance
+static std::unique_ptr<propt> make_satcheck(
+  const optionst &options,
+  bool use_simplifier,
+  message_handlert &message_handler)
+{
+  const auto satcheck_info = parse_satcheck_info(options);
+  std::unique_ptr<propt> satcheck;
+  if(use_simplifier)
+    satcheck = util_make_unique<satcheckt>();
+  else
+    satcheck = util_make_unique<satcheck_no_simplifiert>();
+  satcheck->set_message_handler(message_handler);
+  satcheck->set_random_seed(satcheck_info.random_seed);
```

Comment:
I suggest to check here whether the solver has the feature (`satcheck->has_random_seed()` and output an error message otherwise. Alternatively, at least output a warning in the base class (see below).

Reply:
Warning added.